pkg/options/start.go

```diff
@@ -54,6 +63,37 @@ func resourceByType(resources pipelineResources, restype string) []string {
 	return []string{}
 }
 
+func (taskRunOpts *TaskRunOpts) UseTaskRunFrom(tr *v1beta1.TaskRun, cs *cli.Clients, tname string, taskKind string) error {
+	var (
+		trUsed *v1beta1.TaskRun
+		err    error
+	)
+	if taskRunOpts.Last {
+		trUsed, err = task.LastRun(cs, tname, taskRunOpts.CliParams.Namespace(), taskKind)
+		if err != nil {
+			return err
+		}
+	} else if taskRunOpts.UseTaskRun != "" {
+		trUsed, err = tractions.Get(cs, taskRunOpts.UseTaskRun, metav1.GetOptions{}, taskRunOpts.CliParams.Namespace())
+		if err != nil {
+			return err
+		}
+	}
+
+	if trUsed.Spec.TaskRef.Kind != v1beta1.TaskKind(taskKind) {
+		return fmt.Errorf("%s doesn't belong to %s of kind %s", trUsed.ObjectMeta.Name, tname, taskKind)
+	}
```

Comment:
This check will produce the error when we will try to use the `TaskRun` of `ClusterTask/Task` for `tkn task/clustertask start --use-taskrun`
```
$ tkn t start read-task --prefix-name taskrun --use-taskrun ctaskrun-st9mq
Error: ctaskrun-st9mq doesn't belong to read-task of kind Task
```
Please let me know if this looks good to you 
/cc @danielhelfand

Reply:
I think this should be fine for now. One thing we should maybe discuss later on is if `--use-taskrun` should allow use of TaskRuns that were not created by the Task or ClusterTask used in start. As of now, and it seems since this feature was put in place, that can still happen. However, since this is the current behavior, I think it is fine to leave for now.